kclvm/sema/src/core/symbol.rs

```diff
@@ -0,0 +1,389 @@
+                    .get(schema_ref)
+                    .unwrap()
+                    .parent_schema
+                    .map(|symbol_ref| self.resolve_symbol(symbol_ref, package_info));
+            }
+
+            self.schemas.get_mut(schema_ref).unwrap().mixins = self
+                .schemas
+                .get(schema_ref)
+                .unwrap()
+                .mixins
+                .iter()
+                .map(|symbol_ref| self.resolve_symbol(*symbol_ref, package_info))
+                .collect();
+        }
+    }
+
+    pub fn alloc_package_symbol(&mut self, pkg: PackageSymbol) -> SymbolRef {
+        let id = self.packages.insert(pkg);
+        SymbolRef {
+            id,
+            kind: SymbolKind::Package,
+        }
+    }
+
+    pub fn alloc_schema_symbol(&mut self, schema: SchemaSymbol) -> SymbolRef {
+        let id = self.schemas.insert(schema);
+        SymbolRef {
+            id,
+            kind: SymbolKind::Schema,
+        }
+    }
+
+    pub fn alloc_unresolved_symbol(&mut self, unresolved: UnresolvedSymbol) -> SymbolRef {
+        let id = self.unresolved.insert(unresolved);
+        SymbolRef {
+            id,
+            kind: SymbolKind::Unresolved,
+        }
+    }
+
+    pub fn alloc_type_alias_symbol(&mut self, alias: TypeAliasSymbol) -> SymbolRef {
+        let id = self.type_aliases.insert(alias);
+        SymbolRef {
+            id,
+            kind: SymbolKind::TypeAlias,
+        }
+    }
+
+    pub fn alloc_rule_symbol(&mut self, rule: RuleSymbol) -> SymbolRef {
+        let id = self.rules.insert(rule);
+        SymbolRef {
+            id,
+            kind: SymbolKind::Rule,
+        }
+    }
+
+    pub fn alloc_attribute_symbol(&mut self, attribute: AttributeSymbol) -> SymbolRef {
+        let id = self.attributes.insert(attribute);
+        SymbolRef {
+            id,
+            kind: SymbolKind::Attribute,
+        }
+    }
+}
+#[allow(unused)]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum SymbolKind {
+    Schema,
+    Function,
+    Attribute,
+    Value,
+    Package,
+    TypeAlias,
+    Unresolved,
+    Rule,
+    Dummy,
+}
+#[allow(unused)]
+#[derive(Debug, Clone, Copy)]
+pub struct SymbolRef {
+    id: generational_arena::Index,
+    kind: SymbolKind,
+}
+
+impl SymbolRef {
+    pub fn dummy_symbol() -> Self {
+        Self {
+            id: generational_arena::Index::from_raw_parts(0, 0),
+            kind: SymbolKind::Dummy,
+        }
+    }
+}
+
+impl SymbolRef {
+    pub fn get_kind(&self) -> SymbolKind {
+        self.kind
+    }
+
+    pub fn get_id(&self) -> generational_arena::Index {
+        self.id
+    }
+}
+#[allow(unused)]
+#[derive(Debug)]
+pub struct SchemaSymbol {
+    pub(crate) name: String,
+    pub(crate) start: Position,
+    pub(crate) end: Position,
+    pub(crate) owner: SymbolRef,
+
+    pub(crate) parent_schema: Option<SymbolRef>,
+    pub(crate) mixins: Vec<SymbolRef>,
+    pub(crate) attributes: IndexMap<String, SymbolRef>,
+}
```

Comment:
\+ schema_parameters
\+ doc

Reply:
The addition of symbol metadata will be done in the new resolver, and the namer is only responsible for finding symbols that can be accessed across packages.